community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -404,6 +406,29 @@ def sync_slurm():
     for status, nodes in node_statuses.items():
         do_node_update(status, nodes)
 
+def fr_sync(nodes:List[str])-> Optional[List[str]]:
+    if not nodes:
+        return
+
+    fr_nodes = []
+    for i in range(len(nodes)):
+        if (lookup().node_is_fr(nodes[i])):
+            fr_nodes.append(nodes[i])
+
+    fr = lookup().future_reservation(lookup().node_nodeset(fr_nodes[0])) #we only need one fr for the whole nodeset
+    shared_status = lookup().slurm_node(fr_nodes[0]).base #status should be shared amongst these nodes
```

Comment:
> status should be shared amongst these nodes

This is false assumption, node state can change independently for multiple reasons.

Reply:
Refactored to move away from this assumption, fr nodes only have special actions if they are outside of the bounds of the res, other we pass and let slurmsync bring handle their actions.